core/orchestrator.go

```diff
@@ -552,6 +552,9 @@ func (n *LivepeerNode) transcodeSeg(ctx context.Context, config transcodeConfig,
 	start := time.Now()
 	tData, err := transcoder.Transcode(ctx, md)
 	if err != nil {
+		if _, ok := err.(UnrecoverableError); ok {
```

Comment:
There is no problem with B->O, when the connection is closed, B will retry the segment with a different O.

And yes, B treats this error as retryable, but my understanding is that it's a good thing. Because the problem is not with the segment, but with O, so we want to retry the segment with a different O.

Reply:
OK cool, let's see how this performs then. Yeah, on the case we saw the error was returned for a `record-tester` stream which certainly has well-behaved segments. Hopefully the problem hadn't been caused by a previous segment from another stream that left the orchestrator stuck on that error, tho I think the retries won't be a big problem anyway. 

If we treated the error as unretriable, it also meant that the stream would be stuck on that orchestrator anyway so we would require further changes in the O balancing logic of the Bs, so let's go with the simpler solution for now.